src/generators/Minecraft.res

```diff
@@ -118,36 +124,83 @@ module Cuboid = {
       rotate: 0.0,
     }
 
-    // When a face is flipped both vertically and horizontally, this is the same as rotating 180 degrees.
-    let flip = (face: t, flip: Generator_Texture.flip) => {
-      let (newFlip, newRotate) = switch (face.flip, flip) {
-      | (#None, #None) => (#None, face.rotate)
-      | (#None, #Vertical) => (#Vertical, face.rotate)
-      | (#None, #Horizontal) => (#Horizontal, face.rotate)
-      | (#Vertical, #None) => (#Vertical, face.rotate)
-      | (#Vertical, #Vertical) => (#None, face.rotate)
-      | (#Vertical, #Horizontal) => (#None, face.rotate +. 180.0)
-      | (#Horizontal, #None) => (#Horizontal, face.rotate)
-      | (#Horizontal, #Vertical) => (#None, face.rotate +. 180.0)
-      | (#Horizontal, #Horizontal) => (#None, face.rotate)
-      }
+    let rotateOnAxis = ({rectangle, flip, rotate}: t, axis: (float, float), r: float) => {
+      let rad = r *. Js.Math._PI /. 180.0
+      let (cos, sin) = (Js.Math.cos(rad), Js.Math.sin(rad))
+      let (x, y, w, h) = rectangle
+      let (x0, y0) = axis
+      let (x1, y1) = (toFloat(x) -. x0, toFloat(y) -. y0)
+      let (x2, y2) = (x1 *. cos -. y1 *. sin, x1 *. sin +. y1 *. cos)
+      let (x3, y3) = (toInt(x2 +. x0), toInt(y2 +. y0))
+
       {
-        rectangle: face.rectangle,
-        flip: newFlip,
-        rotate: newRotate,
+        rectangle: (x3, y3, w, h),
+        flip: flip,
+        rotate: addAngle(rotate, r),
       }
     }
 
     let rotate = ({rectangle, flip, rotate}: t, r: float) => {
       let (x, y, w, h) = rectangle
+      let r = addAngle(r, 0.0)
+
+      let f =
```

Comment:
Hey @NinjolasNJM let's see if we can progress this PR.

It's been a while since I looked at it and I'm having difficulty following it.

Could you please add comments to explain as much as you can. Imagine someone has never seen this code before and is looking to understand it.

Also have a think about variable names and see if each one is meaningful, and consider adding more variables to help describe what's happening.

If you're not already aware of it, there is a good book called Clean Code that's worth a read. Someone created a summary of it here:

https://github.com/JuanCrg90/Clean-Code-Notes